providers/src/airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -655,6 +668,31 @@ def execute_sync(self, context: Context):
         if self.do_xcom_push:
             return result
 
+    @tenacity.retry(
+        wait=tenacity.wait_exponential(max=15),
+        retry=tenacity.retry_if_exception_type(PodCredentialsExpiredFailure),
+        reraise=True,
+    )
+    def await_init_containers_completion(self, pod: k8s.V1Pod):
+        try:
+            if self.get_init_containers_logs:
```

Comment:
I think you don't need to check the option setting again here if the calling method checked before calling this function

Reply:
Right, I agree. 

But it would be better to remove self.get_init_containers_logs before calling await_init_containers_completion to mirror the behavior of await_pod_start more closely."

It sounds like you’re refining the control flow for better consistency - feels like a solid approach for streamlining the logging functionality.